drivers/dai/intel/ssp/ssp.c

```diff
@@ -34,6 +34,11 @@ LOG_MODULE_REGISTER(LOG_DOMAIN);
 #define DAI_DIR_CAPTURE 1
 #define SSP_ARRAY_INDEX(dir) dir == DAI_DIR_RX ? DAI_DIR_CAPTURE : DAI_DIR_PLAYBACK
 
+/* Is dynamic clock gating a must? */
+#if DT_INST_PROP_BY_PHANDLE(0, settings, force_dynamic_clock_gating)
+#define SSP_CLK_FORCE_DYNAMIC_CLOCK_GATING 1
+#endif
```

Comment:
Store this in `struct dai_config`

Reply:
I'm not sure I get why... this configuration only defines if some code is going to be compiled or not - why store it in runtime?

Reply:
To avoid hard-coding the instance number

Reply:
But the hard-coded instance number is zero, which is kinda safe, no? Any instance could be linked to the information, using the first one seems reasonable - I think I saw that usage elsewhere.
I also wonder if we are trying to satisfy too many constraints here: avoid using DT_NODELABEL directly, avoid hardcoding an instance number. Although I still think those are hardware properties and should be described in DTS, it seems that by moving to Kconfig would allow us workaround those restrictions.

Reply:
Instance numbers are not safe. Hardcoding a nodelabel and hardcoding an instance number are two variations of the same problem.

Reply:
Ok, will see if there's any way to salvage DTS here, and if not, I'll move those to Kconfigs.